html_css/v2/main-course/custom-properties.md

```diff
@@ -0,0 +1,205 @@
+
+That doesn't mean a custom property can *only* be accessed on the selector it was declared for, though. The scope includes the selector the custom property was declared for as well as the selector's descendants. Let's take a look at a modified version of the above example:
+
+~~~html
+<div class='cool-div'>
+  <p class='cool-paragraph'>Lorem ipsum dolor sit amet.</p>
+</div>
+~~~
+~~~css
+.cool-div {
+  --main-bg: red;
+}
+
+.cool-paragraph {
+  background-color: var(--main-bg);
+}
+~~~
+
+Even though we declared our custom property on our `.cool-div` selector, we can still access it on its descendant `.cool-paragraph` selector. Thanks, dad! Er, thanks `cool-div`!
+
+#### The `:root:` Selector
+
+While there may be times where you will want to limit the scope of a custom property, you may want to be able to use other custom properties on many, unrelated selectors. One work around would be declaring the same custom property on a bunch of selectors, but that defeats one of the purposes of using custom properties in the first place (the ease of changing multiple instances of a value at once).
+
+A better solution is declaring those custom properties on the `:root` selector, which is essentially the same thing as the `html` selector except it has a higher specificity.
+
+~~~html
+<p class='cool-paragraph'>Lorem ipsum dolor sit amet.</p>
+
+<p class='exciting-paragraph'>Lorem ipsum dolor sit amet.</p>
+~~~
+~~~css
+:root {
+  --main-color: red;
+}
+
+.cool-paragraph {
+  color: var(--main-color);
+}
+
+.exciting-paragraph {
+  background-color: var(--main-color);
+}
+~~~
+
+By declaring our custom property on the `:root` selector in the example above, we can access it on *any* other valid selector within our CSS file, since any other selector would be considered a descendant of the `:root` selector.
+
+### Themes
+
+Beyond allowing us to access custom properties more globally, the `:root` selector gives us one way to add themes to our pages. Keep in mind that we're not covering every way to add theming in this lesson, as we just want to introduce you to a simple way that you can get a lot of use out of.
+
+~~~html
+<html class='dark'>
+<!-- our boilerplate... -->
+  <body>
+    <div class='themed-element'>Our themed text</div>
+  </body>
+</html>
+~~~
+~~~css
+:root.dark {
+  --color-background: rgb(18, 18, 18);
+  --color-text: rgb(255, 255, 255);
+}
+
+:root.light {
+  --color-background: rgb(255, 255, 255);
+  --color-text: rgb(18, 18, 18);
+}
+
+.themed-element {
+  background-color: var(--color-background);
+  color: var(--color-text);
+}
+~~~
+
+We just created some quick themes! So what is actually going on here? First, we added a `class` attribute on our `html` element in the HTML file so that we can have a more specific selector to add custom properties onto, and so that our page has a default theme.
```

Comment:
it would be pretty wicked to throw in a codepen embed with a button to toggle the themes.

Reply:
Was gonna ask about the permissions for TOP's Codepen, since I think at least one Codepen embed would work really well on this lesson.